### PR TITLE
Indexify-hotfix

### PR DESCRIPTION
**Pull request by mloubout**
Small PR to fix `indexify` for Sub/Conditional dimensions

Added two test for FD in combination with Sub/Conditional dimensions.

### CODE REVIEW

File: devito/types.py [hunk 1]
```diff
@@ -622,9 +622,18 @@ def indexify(self, indices=None):
...
-        # Only replace spacing -> 1 if used as index
-        subs = dict([(i.spacing, 1) for i in self.indices if
-                     any(i in a.args for a in self.args)])
+
+        # Get spacing symbols for replacement
+        spacings = [i.spacing for i in self.indices]
+
+        # Only keep the ones used as indices.
+        spacings = [s for i, s in enumerate(spacings)
+                    if tuple(s.free_symbols)[0] in self.args[i].free_symbols]
```
Comment (FabioLuporini) [inline devito/types.py]: why u taking the first entry in ``tuple(s.free_symbols)``?
isn't ``tuple(s.free_symbols)[0] == s`` ?

File: tests/test_dimension.py [hunk 2]
```diff
@@ -379,6 +379,29 @@ def test_subdimleft_notparallel(self):
         assert np.all(u.data[0, thickness:] == 0)
         assert np.all(u.data[0, :, thickness+12:] == 0)
 
+    def test_subdim_fd(self):
+        """
+        Test that the FD shortcuts are handled correctly qith subdimensions
```
Comment (FabioLuporini) [inline tests/test_dimension.py]: typo: with
subdimensions -> SubDimensions

File: tests/test_dimension.py [hunk 3]
```diff
@@ -437,6 +460,33 @@ def test_spacial_subsampling(self):
         # Verify that u2[x,y]= u[2*x, 2*y]
         assert np.allclose(u.data[:-1, 0:-1:2, 0:-1:2], u2.data[:-1, :, :])
 
+    def test_subsampled_fd(self):
+        """
+        Test that the FD shortcuts are handled correctly qith subdimensions
```
Comment (FabioLuporini) [inline tests/test_dimension.py]: same as before

File: devito/types.py (hunk 1 above)
Comment (mloubout) [reply devito/types.py]: `s.free_symbols` is a set

Comment (FabioLuporini) [reply devito/types.py]: ah, then use intersection

**Pull request merged**